s3torchconnector/tst/e2e/test_distributed_training.py

```diff
@@ -0,0 +1,183 @@
+#  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#  // SPDX-License-Identifier: BSD
+
+from __future__ import annotations
+
+from collections import Counter
+from itertools import product
+from typing import Callable, TYPE_CHECKING
+
+import pytest
+import torch.multiprocessing as mp
+from torch.utils.data import DataLoader, DistributedSampler
+
+from s3torchconnector import S3IterableDataset, S3MapDataset
+
+if TYPE_CHECKING:
+    from .conftest import BucketPrefixFixture, BucketPrefixData
+
+
+from test_common import _get_start_methods, _read_data, _set_start_method
+
+
+start_methods = _get_start_methods()
+
+import torch.distributed as dist
+
+
+def setup(unique_port, rank, world_size):
+    dist.init_process_group(
+        backend="gloo",
```

Comment:
Shall we test other backend too, such as NCCL?

Reply:
There shouldn't be a difference in type of used backed, but more importantly NCCL backed requires GPU, which is not available on hosts used for running tests.